aws/resource_aws_dms_endpoint_test.go

```diff
@@ -234,18 +237,22 @@ func TestAccAwsDmsEndpoint_Elasticsearch_FullLoadErrorPercentage(t *testing.T) {
 func TestAccAwsDmsEndpoint_Kafka_Broker(t *testing.T) {
 	resourceName := "aws_dms_endpoint.test"
 	rName := acctest.RandomWithPrefix("tf-acc-test")
+	brokerPrefix := "ec2-12-345-678-901.compute-1"
+	brokerPort1 := "2345"
+	brokerPort2 := "3456"
 
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
 		CheckDestroy: dmsEndpointDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: dmsEndpointKafkaConfigBroker(rName, "ec2-12-345-678-901.compute-1.amazonaws.com:2345"),
+				Config: dmsEndpointKafkaConfigBroker(rName, brokerPrefix, brokerPort1),
 				Check: resource.ComposeTestCheckFunc(
 					checkDmsEndpointExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "kafka_settings.#", "1"),
-					resource.TestCheckResourceAttr(resourceName, "kafka_settings.0.broker", "ec2-12-345-678-901.compute-1.amazonaws.com:2345"),
+					resource.TestMatchResourceAttr(resourceName, "kafka_settings.0.broker", regexp.MustCompile(
+						strings.ReplaceAll(strings.ReplaceAll(fmt.Sprintf(`^%s.[^:]+:%s`, brokerPrefix, brokerPort1), "-", `\-`), ".", `\.`))),
```

Comment:
Similarly, we may want to create a "WithPort" equivalent of `testAccMatchResourceAttrRegionalHostname` to simplify testing here 😄  e.g.

```go
// testAccMatchResourceAttrRegionalHostnameWithPort ensures the Terraform state regexp matches a formatted DNS hostname with region, partition DNS suffix, and given port
func testAccMatchResourceAttrRegionalHostnameWithPort(resourceName, attributeName, serviceName string, hostnamePrefixRegexp *regexp.Regexp, port int) resource.TestCheckFunc {
	return func(s *terraform.State) error {
		hostnameRegexpPattern := fmt.Sprintf("%s\\.%s\\.%s\\.%s:%d$", hostnamePrefixRegexp.String(), serviceName, testAccGetRegion(), testAccGetPartitionDNSSuffix(), port)

		hostnameRegexp, err := regexp.Compile(hostnameRegexpPattern)

		if err != nil {
			return fmt.Errorf("Unable to compile hostname regexp (%s): %s", hostnameRegexp, err)
		}

		return resource.TestMatchResourceAttr(resourceName, attributeName, hostnameRegexp)(s)
	}
}
```

Since the hostname is fake anyways, we can inject the region in there (which hey, [matches anything outside us-east-1 anyways](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-dns.html#vpc-dns-hostnames)). This is what we get for the API not validating things upfront and us utilizing that lack of validation to not spin up a real instance to use the attribute reference. 😛